dali/kernels/mask/grid_mask_gpu.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_MASK_GRID_MASK_GPU_H_
+#define DALI_KERNELS_MASK_GRID_MASK_GPU_H_
+
+#include <cuda_runtime.h>
+#include <vector>
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+
+namespace detail {
+
+struct Args {
+  int width, height, channels;
+  float ca, sa, sx, sy, ratio;
+};
+
+template <typename T>
+__global__ void GridMaskKernel(T *output, const T *input, const Args args) {
+  int x = 32 * blockIdx.x + threadIdx.x;
+  int y = 32 * blockIdx.y + threadIdx.y;
```

Comment:
While horizontal warp-sized blocks are a good idea, the vertical size should be configurable and possibly tuned. General experience suggests that 256 or 512 threads per block yields better performance - in any case, this should be tuned.
Having said all that, the very first thing to address is batch processing.

Reply:
this is now handled by BlockSetup, but I'm not sure if I'm using it efficiently